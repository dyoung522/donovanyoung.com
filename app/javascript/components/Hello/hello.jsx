import * as React from "react";
import PropTypes from "prop-types";

const Hello = ({ greet, railsVersion }) => {
  const reactVersion = React.version.match(/\d+\.\d+\.\d+/);

  return (
    <div className="flex justify-center">
      <div className="text-center items-center">
        <p className="py-16 text-2xl text-red-500">Hello {greet || "World"}!</p>
        <p>
          ReactJS v{reactVersion} Running in Rails {railsVersion}
        </p>
      </div>
    </div>
  );
};

Hello.propTypes = {
  greet: PropTypes.string,
  railsVersion: PropTypes.string,
};

export default Hello;
