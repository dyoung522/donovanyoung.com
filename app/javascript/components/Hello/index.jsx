import * as React from "react";
import PropTypes from "prop-types";

const Hello = ({ name }) => {
  return (
    <div className="h-screen flex justify-center">
      <div className="pt-16 text-2xl text-red-500">Hello {name || "World"} from ReactJS in Rails 7!</div>
    </div>
  );
};

Hello.propTypes = {
  name: PropTypes.string,
};

export default Hello;
