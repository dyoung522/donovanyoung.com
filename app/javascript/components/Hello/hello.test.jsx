import React from "react";
import renderer from "react-test-renderer";
import Hello from "./";

it("Hello renders without error", () => {
  const tree = renderer.create(<Hello />).toJSON();

  expect(tree).toMatchSnapshot();
});
