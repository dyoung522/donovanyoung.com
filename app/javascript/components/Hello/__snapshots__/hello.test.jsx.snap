// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders without error 1`] = `
<div
  className="h-screen flex justify-center"
>
  <div
    className="pt-16 text-2xl text-red-500"
  >
    Hello 
    World
     from ReactJS in Rails 7!
  </div>
</div>
`;
